### PR TITLE
fix the progress bar

### PR DESCRIPTION
**Pull request by juneluo**
# Description
Fixes bug priority low 16.
Currently the color and value of progress bar shown in the summary bar and task list are incorrect and inconsistent, they should reflect the percentage of the total committed hours for the week that have been completed, instead of exact hour counts.
**2nd commit note:** Changes made for the leaderboard has been undo and the info description has been updated.

## Mainly changes explained:
1. Color: 0-20%: Red, 20-40%: Orange, 40-60% hrs: Green, 60-80%: Blue, 80-100%:Indigo, and Equal or More than 100%: Purple. The value of the progress bar is set to reflect its percentage.
2. Update progress bars in the summary bar, leaderboard and team member tasks list.
Screen Shot1
Screen Shot3

## How to test:
1. check into current branch
4. do `npm install` and `...` to run this PR locally
5. go to dashboard→ Tasks

### CODE REVIEW

**Comment by one-community** [comment]
Shoot, just realized my description for this is not correct. The LEADERBOARD as it is, is correct. That board is supposed to have the colors based on specific hours, exactly as they are. The rest though need to be updated as described. Please remove any edits to the leaderboard, I don't want that to change.

File: src/components/LeaderBoard/Leaderboard.jsx [hunk 1]
```diff
@@ -116,8 +116,8 @@ const LeaderBoard = ({
                 system, how many volunteer hours they are collectively committed to, and how many
                 tangible and total hours they have completed. The color and length of that bar
                 changes based on what percentage of the total committed hours for the week have been
-                completed: >10%: Red, 10-50%: Orange, 50-60% hrs: Green, 60-70%: Blue, 70-80%:
-                Indigo, 80-100%: Violet, and More than 100%: Purple.
+                completed: 0-20%: Red, 20-40%: Orange, 40-60% hrs: Green, 60-80%: Blue, 80-100%:
```
Comment (LeeSingChanYau) [inline src/components/LeaderBoard/Leaderboard.jsx:119]: Please undo the changes in LeaderBoard

File: src/components/SummaryBar/SummaryBar.jsx [hunk 2]
```diff
@@ -169,38 +170,6 @@ const SummaryBar = props => {
   //   await this.props.getWeeklySummaries(this.props.currentUser.userid);
   //   const { weeklySummariesCount } = this.props.summaries;}
 
-  const getBarColor = hours => {
-    if (hours < 5) {
-      return 'red';
-    }
-    if (hours < 10) {
-      return 'orange';
```
Comment (LeeSingChanYau) [inline src/components/SummaryBar/SummaryBar.jsx:177]: Please confirm with Jae if he wants the summary to be changed.

File: src/utils/effortColors.js [hunk 3]
```diff
@@ -1,27 +1,27 @@
...
-  if (_.inRange(effort, 30, 40)) color = 'super'; //indigo
-  if (_.inRange(effort, 40, 50)) color = 'awesome'; //violet
+export const getcolor = (effort,commit) => {
+  let color = 'white';
+  let percentage = 0;
+  if (commit>0){
+    percentage = Math.round(effort*100/commit);
+  }
+  if (_.inRange(percentage, 0, 20)) color = 'danger'; //red
+  if (_.inRange(percentage, 20, 40)) color = 'orange'; //orange
```
Comment (LeeSingChanYau) [inline src/utils/effortColors.js:9]: Looks good! No need to undo.

File: src/components/TeamMemberTasks/TeamMemberTasks.jsx [hunk 4]
```diff
@@ -15,7 +15,7 @@ import { TaskDifferenceModal } from './components/TaskDifferenceModal';
 import { getTeamMemberTasksData } from './selectors';
 import { getUserProfile } from '../../actions/userProfile';
 import './style.css';
-import { getcolor } from '../../utils/effortColors';
+import { getcolor, getprogress } from '../../utils/effortColors';
 import { fetchAllManagingTeams } from '../../actions/team';
 import EffortBar from 'components/Timelog/EffortBar';
 import TimeEntry from 'components/Timelog/TimeEntry';
```
Comment (LeeSingChanYau) [inline src/components/TeamMemberTasks/TeamMemberTasks.jsx:21]: Looks good! No need to undo.

**Review by LeeSingChanYau — COMMENTED** [review]
Double check if SummaryBar needs to be updated. Changes in Timelog and TeamMemberTasks look good!

File: src/utils/effortColors.js [hunk 5]
```diff
@@ -24,4 +49,4 @@ export const getprogress = (effort) => {
   return progress;
 };
 
-export default getcolor;
+export default {getProgressColor,getcolor};
```
Comment (RJBornToWin) [inline src/utils/effortColors.js:52]: 1. Can you please include comment on why/where it is being used (eg. Leaderboard, tasks, etc) ?  Because its confusing to see 2 different functions names with similar functionality. 
2. Is the export statement at the end partially updated or is it intentional ? I am thinking if "{getProgressColor,getcolor}" should be "{ getProgressColor, getProgressValue }" ?

**Review by RJBornToWin — COMMENTED** [review]
Neat work. Added 2 suggestions for better readability.

**Review by RJBornToWin — APPROVED** [review]
Tested locally for summary bar color changes. Approve the code changes

**Comment by one-community** [comment]
This is excellent! I have one requested update/change. For THE TASK BARS please invert the colors and update the percentages to these: Color: 0-20%: Purple 20-40%: Indigo, 40-60%: Blue, 60-80%: Green, 80-90%: Orange, 90-100%: this color #CC3366 and Equal to or More than 100%: this bright red: #FF3300 

Again, this is 100% my mistake asking for this incorrectly. Updating the bugs list with this new request.

**Comment by one-community** [comment]
It looks like I didn't create this bug, so that's where I think the confusion came from. I thought these were all working well enough how they were. They'll be better than before though thanks to this work on them, so thank you. Please make these additional requested changes Jianjun.
